tests/unit_tests/report/test_project_report.py

```diff
@@ -0,0 +1,186 @@
+        self.assertEqual(mock_print_html.call_count, 2)
+
+    @patch('shapash.report.project_report.print_md')
+    def test_display_general_information_1(self, mock_print_html):
+        report = ProjectReport(
+            explainer=self.xpl,
+            metadata_file=os.path.join(current_path, '../../data/metadata.yaml')
+        )
+        report.display_metadata_information()
+        self.assertTrue(mock_print_html.called)
+
+    @patch('shapash.report.project_report.print_md')
+    def test_display_model_information_1(self, mock_print_md):
+        report = ProjectReport(
+            explainer=self.xpl,
+            metadata_file=os.path.join(current_path, '../../data/metadata.yaml')
+        )
+        report.display_model_information()
+        self.assertTrue(mock_print_md.called)
+
+    def test_display_dataset_analysis_1(self):
+        report = ProjectReport(
+            explainer=self.xpl,
+            metadata_file=os.path.join(current_path, '../../data/metadata.yaml'),
+            x_train=self.df[['x1', 'x2']],
+        )
+        report.display_dataset_analysis()
+
+    def test_display_dataset_analysis_2(self):
+        report = ProjectReport(
+            explainer=self.xpl,
+            metadata_file=os.path.join(current_path, '../../data/metadata.yaml'),
+        )
+        report.display_dataset_analysis()
+
+    def test_display_model_explainability(self):
+        report = ProjectReport(
+            explainer=self.xpl,
+            metadata_file=os.path.join(current_path, '../../data/metadata.yaml'),
+        )
+        report.display_model_explainability()
+
+    @patch('shapash.report.project_report.logging')
+    def test_display_model_performance_1(self, mock_logging):
+        """
+        No y_test given
+        """
+        report = ProjectReport(
+            explainer=self.xpl,
+            metadata_file=os.path.join(current_path, '../../data/metadata.yaml'),
+        )
+        report.display_model_performance()
+        mock_logging.info.assert_called_once()
+
+    @patch('shapash.report.project_report.logging')
+    def test_display_model_performance_2(self, mock_logging):
+        report = ProjectReport(
+            explainer=self.xpl,
+            metadata_file=os.path.join(current_path, '../../data/metadata.yaml'),
+            y_test=self.df['y'],
+            config=dict(metrics={'mse': 'sklearn.metrics.mean_squared_error'})
+        )
+        report.display_model_performance()
+        self.assertEqual(mock_logging.call_count, 0)
+
+    @patch('shapash.report.project_report.logging')
+    def test_display_model_performance_3(self, mock_logging):
+        """
+        No metrics given in ProjectReport
+        """
+        report = ProjectReport(
+            explainer=self.xpl,
+            metadata_file=os.path.join(current_path, '../../data/metadata.yaml'),
+            y_test=self.df['y'],
+        )
+        report.display_model_performance()
+        mock_logging.info.assert_called_once()
```

Comment:
Have you check or test to verify columns between train and test ?

Reply:
Unfortunately this cannot be done easily as the function does not return any dataframe but only some HTML objects which can hardly be inspected